src/components/icon/js/iconDirective.js

```diff
@@ -247,19 +256,135 @@ function mdIconDirective($mdIcon, $mdTheming, $mdAria, $sce) {
             element.append(svg);
           });
         }
-
       });
     }
 
-    function parentsHaveText() {
+    function checkAriaAttributes(domAttrs) {
+      return domAttrs['aria-label'] || domAttrs['aria-labelledby'] || domAttrs['aria-describedby'];
+    }
+
+    function performAriaCheck(domElement) {
+      if (!checkAriaAttributes(domElement.attributes)) {
+        return false;
+      }
+      /* Perform role blacklist check */
+      if (domElement.attributes.role) {
```

Comment:
Please move this to the aria utils. This complexity should not be in the iconDirective.